src/Acuminator/Acuminator.Vsix/Commands/VSCommandBase.cs

```diff
@@ -45,48 +48,53 @@ internal abstract class VSCommandBase
 		/// Adds our command handlers for menu (commands must exist in the command table file)
 		/// </summary>
 		/// <param name="package">Owner package, not null.</param>
-		protected VSCommandBase(Package package, int commandID, Guid? customCommandSet = null)
+		protected VSCommandBase(AsyncPackage package, OleMenuCommandService commandService, int commandID, Guid? customCommandSet = null)
 		{
 			package.ThrowOnNull(nameof(package));
+			commandService.ThrowOnNull(nameof(commandService));
 
 			Package = package;
 			Guid commandSet = customCommandSet ?? DefaultCommandSet;
-			OleMenuCommandService commandService = ServiceProvider.GetService<IMenuCommandService, OleMenuCommandService>();
 
 			if (commandService != null)
 			{
 				var menuCommandID = new CommandID(commandSet, commandID);
-				var menuItem = new OleMenuCommand(CommandCallback, menuCommandID);
-				menuItem.BeforeQueryStatus += QueryFormatButtonStatus;
+				var menuItem = new OleMenuCommand(CommandCallback, menuCommandID)
+				{
+					// This defers the visibility logic back to the VisibilityConstraints in the .vsct file
+					Supported = false
+				};
+
 				commandService.AddCommand(menuItem);
 			}
 		}
 
+
 		/// <summary>
 		/// Gets the service provider from the owner package.
 		/// </summary>
-		protected IServiceProvider ServiceProvider => Package;
+		protected IAsyncServiceProvider ServiceProvider => Package;
 		
-		protected virtual void QueryFormatButtonStatus(object sender, EventArgs e)
-		{
-			if (!(sender is OleMenuCommand menuCommand))
-				return;
+		//protected virtual void QueryFormatButtonStatus(object sender, EventArgs e)
```

Comment:
Do we need this code? If no I propose to remove it.

Reply:
No, we don't but I removed it in in next pull request - because it is directly related to the commands visibility